### ISSUE TITLE
Single quotes (') or double quotes (")

### ISSUE THREAD

**Issue by lukpueh** [secure-systems-lab/code-style-guidelines#4]
In Python single quotes (`'`) and double quotes can be used (`"`) interchangeably to designate strings. While our code-style-guidelines suggest to use triple-double-quotes (`"""`) for docstrings, we don't have a policy for normal strings.

Please participate in this discussion to help us promote quote consistency in our projects!

I'd like to start the discussion with the following suggestion:
- Use single quotes everywhere , unless 
- the string contains single quotes, then use double quotes.

``` python
"""What the foo? """
foo = 'bar'
baz = {'foo': foo}
if baz['foo'] == 'bar':
  print("foo's value is 'bar'")
```

**Comment by vladimir-v-diaz** [comment]
I prefer to use only double quotes for the triple-double-quotes of docstrings and single quotes everywhere else, but wouldn't mind your proposed exception.

What if for some reason the string must contain double quotes?
"foo's value is "bar""

Do we escape it in this case?  I do understand that you'd like to avoid escape sequences in strings.

**Comment by aaaaalbert** [comment]
I prefer double quotes (solo and triple) everywhere. I use them in handwriting, I use them in other languages if allowed.

It's 90% muscle memory, so I could re-learn and use single quotes just as well, but 10% useful for what I think is still a common case: communicating with the user in plain English.

Following your proposal, `'I\'m sorry Dave, I\'m afraid I can\'t do that'` should obviously use double quotes, not escapes. My point is that I think that this happens frequently enough that the *exception* from the single-quote default would either *become the default*, or cause still lots of visual inconsistency (according to or accidentally ignoring the rules) throughout our code. This is only an epsilon better than the mix of quote styles we have at the moment.

**Comment by aaaaalbert** [comment]
Two random data points to fuel the flame war:
* I hope we agree to not consider [this style](https://stackoverflow.com/questions/56011/single-quotes-vs-double-quotes-in-python#56190): `'English': "There are %(number_of_lights)s lights."`
* For people with a C background, double quotes perhaps [look more string-y](https://stackoverflow.com/questions/3683602/single-quotes-vs-double-quotes-in-c-or-c#3683613).

**Comment by lukpueh** [comment]
I must admit that I loosely have been using the style that you'd rather not consider. Single words are almost like single characters whereas phrases..., right? :)

Anyway, I'm fine with Albert's proposal too (double quotes everywhere), as long as we have some policy.

**Comment by awwad** [comment]
For whatever reason, I’ve been using single quotes essentially everywhere (unless they’re in the string, in which case I either escape or use double quotes). I think Vlad and I more or less have the same approach.

I, too, don’t mind changing, if it’s in the name of a consistent lab style, whether it’s almost always single or almost always double.

**Comment by lukpueh** [comment]
Starting with #21 we mandate Google Style Guide and recommend PEP8, both of which only recommend to pick one quote-style and stick with it, unless it requires a lot of escapes in a given string, then it's okay to use the other.

I think we should state a clear preference for either of them in the `String` section of the new style guide added in #21. FWIW, in the examples of that new guide I consistently use double quotes `"`, but it would be easy to switch, if there is a consensus for single `'`.
